thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/merge/AnomalyTimeBasedSummarizer.java

```diff
@@ -132,5 +136,13 @@ private static void populateMergedResult(MergedAnomalyResultDTO mergedAnomaly,
     mergedAnomaly.setStartTime(currentResult.getStartTime());
     mergedAnomaly.setEndTime(currentResult.getEndTime());
     mergedAnomaly.setCreatedTime(System.currentTimeMillis());
+    // populate current result's property as well, will be used to identify if two anomalies can be merged by comparing their property
+    mergedAnomaly.setProperties(AnomalyDetectionUtils.decodeCompactedPropertyStringToMap(currentResult.getProperties()));
+  }
+
+  // compare if two anomalies have same property when doing anomaly detection, if from same detection configuration then is mergeable
+  private static boolean isMergeable(MergedAnomalyResultDTO anomaly1, MergedAnomalyResultDTO anomaly2){
+    return ((anomaly1.getProperties() == null && anomaly2.getProperties() == null)
+        || (anomaly1.getProperties() != null && anomaly1.getProperties().equals(anomaly2.getProperties()) ));
```

Comment:
I have reduced the "property" to contain only the configurations.  There are two facts that I see when doing integration test:
1. Repeated information are being stored in both anomaly.getProperties() as well as anomaly.getFunction().getProperties(). They actually were storing the same function settings, i.e. signTestPattern:"UP,DOWN". Then I re-designed to put only the one side function configuration into anomaly.getProperties()
2. It is hard to decode test_config (key,value) when do update in merge, we want to use the specific one side configuration rather than the two sided anomaly function property. If we really want to recover tag to lift/shift values, we need anther encoder and decoder for anomaly function config tag.